src/java.base/share/classes/jdk/internal/foreign/abi/NativeEntryPoint.java

```diff
@@ -48,7 +48,29 @@ public class NativeEntryPoint {
     private record CacheKey(MethodType methodType, ABIDescriptor abi,
                             List<VMStorage> argMoves, List<VMStorage> retMoves,
                             boolean needsReturnBuffer, int capturedStateMask,
-                            boolean needsTransition) {}
+                            boolean needsTransition) {
+
+        @Override
+        public boolean equals(Object o) {
+            if (!(o instanceof CacheKey other)) return false;
+
+            return methodType == other.methodType && abi == other.abi && capturedStateMask == other.capturedStateMask
```

Comment:
Is this correct? The old code was using `MethodType::equals`, but now we seem to use `==` instead. I suppose the idea is that method types are interned, and their `equals` method is only really meaningful when checking whether a method type with same shape has already been interned?

Reply:
Yep, equals are only used on primordial method types for the intern table. Identity is significant for invokers sharing.